amgut/lib/data_access/ag_data_access.py

```diff
@@ -230,172 +198,154 @@ def registerHandoutKit(self, ag_login_id, supplied_kit_id):
         bool
             True:  success
             False: insert failed due to IntegrityError
-
-        Notes
-        -----
-        Whatever is passed as kit_password will be added AS IS. This means you
-        must hash the password before passing, if desired.
         """
-        printresults = self.checkPrintResults(supplied_kit_id)
-        if printresults is None:
-            printresults = 'n'
-        # make sure login_id and skid exists
-        sql = """SELECT EXISTS(SELECT *
-                               FROM ag.ag_login
-                               WHERE ag_login_id = %s)"""
-        conn_handler = SQLConnectionHandler()
-        try:
-            exists = conn_handler.execute_fetchone(sql, [ag_login_id])[0]
-        except ValueError:
-            raise ValueError("ag_login_id is not a UUID: %s" % ag_login_id)
-        if not exists:
-            return False
-        sql = """SELECT EXISTS(SELECT *
-                               FROM ag.ag_handout_kits
-                               WHERE kit_id = %s)"""
-        if not conn_handler.execute_fetchone(sql, [supplied_kit_id])[0]:
-            return False
-
-        sql = """
-            DO $do$
-            DECLARE
-                k_id uuid;
-                bc varchar;
-            BEGIN
-                INSERT INTO ag_kit
-                (ag_login_id, supplied_kit_id, kit_password, swabs_per_kit,
-                 kit_verification_code, print_results)
-                SELECT '{0}', kit_id, password, swabs_per_kit,
-                    verification_code, '{1}'
-                    FROM ag_handout_kits WHERE kit_id = %s LIMIT 1
-                RETURNING ag_kit_id INTO k_id;
-                FOR bc IN
-                    SELECT barcode FROM ag_handout_barcodes WHERE kit_id = %s
-                LOOP
-                    INSERT  INTO ag_kit_barcodes
-                        (ag_kit_id, barcode, sample_barcode_file)
-                        VALUES (k_id, bc, bc || '.jpg');
-                END LOOP;
-                DELETE FROM ag_handout_kits WHERE kit_id = %s;
-            END $do$;
-            """.format(ag_login_id, printresults)
-
-        try:
-            conn_handler.execute(sql, [supplied_kit_id] * 3)
-        except psycopg2.IntegrityError:
-            logging.exception('Error on skid %s:' % ag_login_id)
-            return False
-        return True
+        with TRN:
+            printresults = self.checkPrintResults(supplied_kit_id)
+            if printresults is False:
```

Comment:
You don't need to check for this n and False will be equivalent in SQL, as long as they are both enclosed by quotes and being inserted into a boolean column - which is this case.